### PR TITLE
valgrind: disable on darwin

### PR DESCRIPTION
**Pull request by zowoq**
Our macOS SDKs are from 10.12 but our cctools is from 10.14 so I think that basically kills off `valgrind` on darwin as upstream doesn't support it on >= 10.13 and no one seems to be building it on >= 10.14.

> https://valgrind.org/
> 22 June 2020: valgrind-3.16.1 is available. This release supports: .... X86/MacOSX 10.12 and AMD64/MacOSX 10.12.

> [homebrew valgrind](https://github.com/Homebrew/homebrew-core/blob/bc253ed78337eaa551bc5abfb4a3a4753b3c5c8b/Formula/valgrind.rb#L10)
> depends_on maximum_macos: :high_sierra

> [macports valgrind](https://github.com/macports/macports-ports/blob/3125ff6b29f31c7ab1612ddce69a59d7bbb1a4b3/devel/valgrind/Portfile#L123)
>  is not presently compatible with macOS Mojave or newer
___

The `libpsl` commit should be enough to unblock `nixpkgs-unstable`.

### CODE REVIEW

**Comment by zowoq** [comment]
cc @matthewbauer @marsam @vcunat

File: pkgs/development/libraries/libpsl/default.nix [hunk 1]
```diff
@@ -31,7 +31,7 @@ stdenv.mkDerivation rec {
     lzip
     pkgconfig
     python3
-    valgrind
+    (stdenv.lib.optionalString (!stdenv.isDarwin) valgrind)
```
Comment (vcunat) [inline pkgs/development/libraries/libpsl/default.nix:34]: Does this work fine with still passing `"--enable-valgrind-tests"`?

Comment (zowoq) [reply pkgs/development/libraries/libpsl/default.nix:34]: Yes, `doCheck` is already disabled on darwin.

Comment (vcunat) [reply pkgs/development/libraries/libpsl/default.nix:34]: Ah, right.  It looks a bit weird but OK I guess.

**Review by vcunat — APPROVED** [review]

File: pkgs/development/libraries/libpsl/default.nix [hunk 2]
```diff
@@ -31,7 +31,7 @@ stdenv.mkDerivation rec {
...
     libxslt
   ];
```
Comment (marsam) [inline pkgs/development/libraries/libpsl/default.nix:36]:
```suggestion
    libxslt
  ] ++ stdenv.lib.optionals (!stdenv.isDarwin) [ valgrind ];
```
besides that, LGTM

Comment (zowoq) [reply pkgs/development/libraries/libpsl/default.nix:36]: Thats is what I did [first](https://github.com/NixOS/nixpkgs/compare/2c28ff2f00730b8f9a3c1547ad9830287de57191..c635f8f156886ca712ec89c431a50ee9bdea4bda) but it causes unnecessary rebuilds on linux.

Comment (marsam) [reply pkgs/development/libraries/libpsl/default.nix:36]: oh, interesting, I'm not sure why it should cause rebuilds on Linux

**Review by marsam — APPROVED** [review]

File: pkgs/development/libraries/libpsl/default.nix (hunk 2 above)
Comment (vcunat) [reply pkgs/development/libraries/libpsl/default.nix:36]: The order of build inputs is preserved into the build process itself.

Comment (marsam) [reply pkgs/development/libraries/libpsl/default.nix:36]: TIL thanks!

**Comment by zowoq** [comment]
Pushed a cleaned up version of this to staging: https://github.com/NixOS/nixpkgs/commit/d126f0b8c6a8b50786d21ed0da0b7343779a6e00

**Comment by vcunat** [comment]
> should be enough to unblock `nixpkgs-unstable`.

Unfortunately there's another problem: https://hydra.nixos.org/job/nixpkgs/trunk/nix-info-tested.x86_64-darwin

**Pull request merged**